docs/user-guide/masking.ipynb

```diff
@@ -237,6 +246,118 @@
     "In both cases the masks that are applied during the operation are not included in the masks of the output.\n",
     "Masks that are independent of the binning dimension(s) are unaffected and included in the output."
    ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Examples and common patterns using masks\n",
+    "\n",
+    "### Masking where a coordinate is negative\n",
+    "\n",
+    "Example data:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "a = sc.DataArray(\n",
+    "    data=sc.ones(dims=['x'], shape=(6,)),\n",
+    "    coords={'x': sc.linspace('x', -1, 1, 6, unit='m')},\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To mask entries of a data array where a coordinate is negative, create a boolean variable that is true where the coordinate is negative, and set it on the `masks` dictionary:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "a.masks['x is negative'] = a.coords['x'] < sc.scalar(0., unit='m')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Masking nan values\n",
+    "\n",
+    "It is common to mask data where the value of some coordinate is nan.\n",
+    "To mask where a coordinate is `nan`-valued we need to create a boolean variable that is `True` where the coordinate is `nan`.\n",
+    "That is exactly what [`sc.isnan`](/generated/functions/scipp.isnan.html#scipp.isnan) is for:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "a.masks['x is nan'] = sc.isnan(a.coords['x'])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Removing masks\n",
+    "\n",
+    "To remove one or several masks, use the `.drop_masks` method:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "a.masks"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "a.drop_masks(['x is negative'])"
```

Comment:
Um, `drop_masks` should not touch the input but return a now object with the masks dropped. Are you sure this example is good/correct?

Also, why not mention `del`?

Reply:
Johannes is already on holiday so I'll just fix this here...?